qemu/tests/timedrift_check_after_load_vm.py

```diff
@@ -0,0 +1,158 @@
+        if session:
+            session.close()
+        env.unregister_vm(vm.name)
+        vm.destroy(gracefully=False, free_mac_addresses=True)
+        _load_kvm_module_with_kvmclock_periodic_sync("Y")
+
+    def create_gagent():
+        """
+        Create a QemuAgent object to send guest agent command
+        """
+        gagent_serial_type = params["gagent_serial_type"]
+        gagent_name = params["gagent_name"]
+        if gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_VIRTIO:
+            filename = vm.get_virtio_port_filename(gagent_name)
+        elif gagent_serial_type == guest_agent.QemuAgent.SERIAL_TYPE_ISA:
+            filename = vm.get_serial_console_filename(gagent_name)
+        else:
+            raise guest_agent.VAgentNotSupportedError("Not supported serial"
+                                                      "type")
+        return guest_agent.QemuAgent(vm, gagent_name, gagent_serial_type,
+                                     filename, get_supported_cmds=True)
+
+    def run_qmp_cmd(qmp_port, qmp_cmd):
+        """
+        Run a qmp command
+
+        :params qmp_port: the guest qmp port to send qmp command
+        :params qmp_cmd: qmp command
+        """
+        output = qmp_port.send_args_cmd(qmp_cmd)
+        error_context.context("QMP command: '%s' \n Output: '%s'"
+                              % (qmp_cmd, output), logging.info)
+
+    def query_ntp_time():
+        """
+        On guest, use "ntpdate -q {ntp_server}" to query the clock offset
+        """
+        ntp_query_cmd = params["ntp_query_cmd"]
+        output = session.cmd_output_safe(ntp_query_cmd)
+        error_context.context("Command: '%s'  \n Output: '%s'"
+                              % (ntp_query_cmd, output), logging.info)
+        offset = float(re.findall(r"[offset|+|-]\s*(\d+\.\d+)", output)[-1])
+        error_context.context("offset: '%.2f'" % offset, logging.info)
+        exptectd_time_drift = params.get("expected_time_drift", 3)
+        if offset > float(exptectd_time_drift):
+            test.fail("After loadvm, the time drift of guest is too large.")
+
+    vm = setup()
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+    try:
+        error_context.context("Sync guest time with ntp server", logging.info)
+        ntp_cmd = params["ntp_cmd"]
+        output = session.cmd_output_safe(ntp_cmd)
+        error_context.context("Command: '%s'  \n Output: '%s'"
+                              % (ntp_cmd, output), logging.info)
+
+        qmp_ports = vm.get_monitors_by_type('qmp')
+        if qmp_ports:
+            qmp_port = qmp_ports[0]
+        else:
+            test.fail("Incorrect configuration, no QMP monitor found.")
+
+        qmp_savevm_cmd = "savevm tag=sn1"
+        run_qmp_cmd(qmp_port, qmp_savevm_cmd)
+        qmp_loadvm_cmd = "loadvm tag=sn1"
+        run_qmp_cmd(qmp_port, qmp_loadvm_cmd)
+
+        gagent = create_gagent()
+        gagent.set_time()
```

Comment:
> If no @time is specified, then the time to set is read from RTC. However, this may not be supported on all platforms (i.e. Windows). 
This may have a windows limitation, please take care of the drift in windows.

Reply:
@sitoliu Thanks.

There is a bug bz1183874. Windows guest cannot be synced with RTC time using "guest-set-time". After bz1183874 is fixed, this case can be passed.

Reply:
@dinghuiqing Thanks.